tests/testdata/generate.rs

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env -S cargo -Z script
```

Comment:
I think I'd prefer to keep this in `examples` instead of using the nightly stuff, maybe with a docstring to make it clear that it's not actually an example?

Reply:
> I think I'd prefer to keep this in examples instead of using the nightly stuff, maybe with a docstring to make it clear that it's not actually an example?

Sure, WFM.

> I imagine that might also help avoid these certs become stale over time and needing to be regenerated every x months?

The script is using the rcgen default validity period which is insanely large, so at least that won't be a problem. If you don't value the reproducibility we could also just vendor the cert and key without a generator. There's nothing "special" about the certs except the localhost/127.0.0.1 SAN on the EE.

Reply:
> I think I'd prefer to keep this in examples instead of using the nightly stuff, maybe with a docstring to make it clear that it's not actually an example?

Done.

> Doing it inline in the test also makes a lot of sense

I ended up leaving this as an example generator + vendored certs/key because it let us keep the `static` pebble config. Unfortunately there's no way to provide the management cert/key by command-line flags and so making them dynamically complicates the default config where it needs file paths.

It's a special kind of "fun" to suffer your own code as a user years after writing it. Turns out future me is not a fan of some things past me did :zany_face: